overlord/ifacestate/handlers.go

```diff
@@ -1181,3 +1181,64 @@ func (m *InterfaceManager) doGadgetConnect(task *state.Task, _ *tomb.Tomb) error
 	task.SetStatus(state.DoneStatus)
 	return nil
 }
+
+// doHotplugDisconnect creates task(s) to disconnect connections and remove slots in response to hotplug "remove" event.
+func (m *InterfaceManager) doHotplugDisconnect(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	core, err := snapstate.CoreInfo(st)
```

Comment:
this needs to be the system snap, we probably want a helper for this

Reply:
Added in #6146

Reply:
Updated to use the new helper.